MachineLearning/R/mlclassificationknn.R

```diff
@@ -0,0 +1,608 @@
+
+    if(!options[["ready"]]){
+      .evaluationTableClassification(dataset, options, opt, formula, res = NULL, jaspResults)
+    } else {
+
+    formula                      <- .makeformulaClassification(options)
+    # Run the analysis
+    res                         <- .doKNNClassification(dataset, options, opt, formula, jaspResults)
+    res                         <- jaspResults[["res"]]$object
+    # create the evaluation table ##
+    .evaluationTableClassification(dataset, options, opt, formula, res, jaspResults)
+    # Create the confusion table ##
+    if (options[['confusionTable']])
+    	{
+    		if(is.null(jaspResults[["confusionTable"]]))
+    			.confusionTableClassification(dataset, options, res, jaspResults)
+    	}
+    # # Create the predictions table ##
+    if (options[['tablePredictions']])
+      {
+        if(is.null(jaspResults[["predictionsTable"]]))
+          .predictionsTableClassification(options, opt, res, jaspResults)
+      }
+    # Create the distances table ##
+    if (options[['tableDistances']])
+      {
+        if(is.null(jaspResults[["distancesTable"]]))
+          .distancesTableClassification(options, opt, res, jaspResults)
+      }
+    # Create the weights table ##
+    if(options[['tableWeights']])
+    {
+      if(is.null(jaspResults[["weightsTable"]]))
+        .weightsTableClassification(options, opt, res, jaspResults)
+    }
+    # Create the Error vs K plot ##
+    if(options[['plotErrorVsK']])
+    {
+       if(is.null(jaspResults[["plotErrorVsK"]]))
+       {
+       jaspResults[["plotErrorVsK"]] 		<- .plotErrorVsKClassification(dataset, options, opt, formula, res, jaspResults)
+       jaspResults[["plotErrorVsK"]]		$dependOnOptions(c("plotErrorVsK"))
+       jaspResults[["plotErrorVsK"]] 		$position <- 3
+       }
+    }
+  }
+}
+
+.readDataMlKNNclassification <- function(dataset, options){
+  predictors                <- unlist(options['predictors'])
+  predictors                <- predictors[predictors != ""]
+  target                    <- NULL
+  if(!(options[["target"]] == ""))
+    target                  <- options[["target"]]
+  variables.to.read         <- c(predictors, target)
+  if (is.null(dataset)) {
+          dataset <- .readDataSetToEnd(columns.as.numeric=variables.to.read,exclude.na.listwise=variables.to.read)
+  }
+    return(dataset)
+}
+
+.mlKnnClassificationErrorHandling <- function(dataset, options){
+  predictors                <- unlist(options['predictors'])
+  predictors                <- predictors[predictors != ""]
+  target                    <- NULL
+  if(!(options[["target"]] == ""))
+    target                  <- options[["target"]]
+  variables.to.read         <- c(predictors, target)
+  for(i in 1:length(variables.to.read)){
+      errors <- .hasErrors(dataset, perform, type = c('infinity', 'observations'),
+                           all.target = variables.to.read[i],
```

Comment:
doesn't this work?
```suggestion
                           all.target = variables.to.read,
```
So you could skip the loop?